src/main/java/io/mosip/iiitb/odte/lib/MessageBrokerWrapper.java

```diff
@@ -0,0 +1,146 @@
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, config.kafkaBrokers());
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, config.kafkaGroupId());
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                StringDeserializer.class.getName()
+        );
+        props.put(
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                StringDeserializer.class.getName()
+        );
+
+        logger.debug(
+                String.format(
+                        "Listening on %s:%s\nTopics = %s\nGroupId = %s\nBrokers = %s\n",
+                        config.kafkaHostname(),
+                        config.kafkaPort(),
+                        config.kafkaTopic(),
+                        config.kafkaGroupId(),
+                        config.kafkaBrokers()
+                )
+        );
+        return new KafkaConsumer<>(props);
+    }
+
+    private void loopIt() {
+        logger.debug("Started listening to events");
+        while (true) {
+            final ConsumerRecords<String, String> consumerRecords =
+                    this.consumer.poll(Duration.ofMillis(100));
+            if (consumerRecords.count() == 0) {
+                continue;
+            }
+
+            for (ConsumerRecord<String, String> record : consumerRecords) {
+                DecryptedOnDemandTemplateRecord drc = parseOnDemandTemplateRecord(record);
+                if (drc != null) {
+                    EventConsumerResponse evtConsumerResponse = this.odteConsumer.processRecord(drc);
+                    logger.debug("Processed a record and got response = {}", evtConsumerResponse);
+                }
+            }
+            this.consumer.commitAsync();
+        }
+    }
+
+    private DecryptedOnDemandTemplateRecord parseOnDemandTemplateRecord(ConsumerRecord<String, String> record) {
+        String jsonMessage = record.value();
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        KafkaOnDemandMessageDto parsedRecord;
+        try {
+            parsedRecord = objectMapper.readValue(jsonMessage, KafkaOnDemandMessageDto.class);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+
+        KafkaOnDemandMessageDto.KafkaOnDemandMessageEventDataDto messageEventData = parsedRecord.getEvent().getData();
+        String encryptedIdBase64 = messageEventData.getIndividualId();
+        String idType = messageEventData.getIndividualIdType();
+        try {
+            byte[] encryptedId = Base64.getDecoder().decode(encryptedIdBase64);
+            byte[] idBytes = this.rsaCryptoTool.decryptData(encryptedId);
+            String id = new String(idBytes);
+            DecryptedOnDemandTemplateRecord dtr = new DecryptedOnDemandTemplateRecord();
+            dtr.setId(id);
```

Comment:
If the same id is comming multiple times, this looks like it will keep generating the credential request. Not sure if IDA is handling this case, please check on this.